lebedigital/openbis/expstep.py

```diff
@@ -0,0 +1,1464 @@
+
+        else:
+            return meta_df
+
+    def import_from_template(self, path_to_file: str):
+        """Imports the data from the template into the samples metadata. To used with the generated import template
+
+        Args:
+            path_to_file (str): Path to the filled out import template
+        """
+
+        df = pd.read_excel(path_to_file)
+        df.columns = df.columns.str.lower()
+
+        # if some params were left out then the row gets dropped
+        df = df.dropna()
+
+        # not useful for uploading to the datastore only for the user to know what to input
+        df.pop('label')
+        df.pop('description')
+
+        # turn the df columns into a dict
+        meta = dict(zip([par.lower() for par in df.param], df.value))
+        self.metadata = meta
+
+    # The simple static methods will get integrated into their methods later because they are useless to define over pybis itself, self explainatory fucntions otherwise
+    @staticmethod
+    def get_space_names(o: Openbis):
+        spaces_df = o.get_spaces().df
+        return list(spaces_df['code'].values)
+
+    @staticmethod
+    def get_project_names(o: Openbis, space: str):
+        projects_df = o.get_projects(space=space).df
+        return [name.split('/')[-1] for name in list(projects_df['identifier'].values)]
+
+    @staticmethod
+    def get_experiment_names(o: Openbis, space: str, project: str):
+        experiments_df = o.get_experiments(
+            space=space,
+            project=project,
+        ).df
+        return [name.split('/')[-1] for name in list(experiments_df['identifier'].values)]
+
+    # Alias
+    get_collection_names = get_experiment_names
+
+    @staticmethod
+    def get_samples(o: Openbis, space: str, project: str, collection: str, **kwargs):
+
+        if "experiment" in kwargs:
+            kwargs["collection"] = kwargs["experiment"]
+            kwargs.pop("experiment", None)
+
+        return o.get_samples(space=space, project=project, collection=collection)
+
+    # Alias
+    get_objects = get_samples
+
+    @staticmethod
+    def get_sample_names(o: Openbis, space: str, project: str, collection: str, **kwargs):
+
+        if "experiment" in kwargs:
+            kwargs["collection"] = kwargs["experiment"]
+            kwargs.pop("experiment", None)
+
+        samples_df = o.get_samples(
+            space=space,
+            project=project,
+            collection=collection,
+            props=['$name']
+        ).df
+        samples_df.columns = samples_df.columns.str.upper()
+        sample_codes = [name.split('/')[-1]
+                        for name in list(samples_df['IDENTIFIER'].values)]
+        sample_names = list(samples_df['$NAME'].values)
+        return [f'{code} ({name})' for code, name in zip(sample_codes, sample_names)]
+
+    get_object_names = get_sample_names
+
+    @staticmethod
+    def get_sample_codes(o: Openbis, space: str, project: str, collection: str, **kwargs):
```

Comment:
see above, I do not know what a sample_code is

Reply:
Hopefully i will have the documentation set up soon where i will write the explanations for these names. I have written you what the code in is in a different comment.